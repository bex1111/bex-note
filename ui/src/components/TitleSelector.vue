<script setup>
const selectedNote = defineModel();
defineProps({
  notes:Array
})

</script>

<template>
  <prime-select
      v-model="selectedNote"
      :options="notes"
      optionLabel="title"
      optionValue="title"
      filter
      placeholder="Select a note"
      fluid
  />
</template>

<style scoped>

</style>