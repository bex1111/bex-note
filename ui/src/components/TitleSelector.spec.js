import { mount } from '@vue/test-utils';
import TitleSelector from './TitleSelector.vue';

describe('TitleSelector.vue', () => {
  it('matches snapshot', () => {
    const mockNotes = [
      { title: 'Note 1' },
      { title: 'Note 2' },
      { title: 'Folder/Nested Note' }
    ];

    const wrapper = mount(TitleSelector, {
      props: {
        notes: mockNotes,
        modelValue: ''
      },
      global: {
        stubs: {
          'prime-select': {
            template: '<div class="prime-select-stub" :options="options" :optionLabel="optionLabel" :optionValue="optionValue" :filter="filter" :placeholder="placeholder" :fluid="fluid">{{ modelValue }}</div>',
            props: ['modelValue', 'options', 'optionLabel', 'optionValue', 'filter', 'placeholder', 'fluid']
          }
        }
      }
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
