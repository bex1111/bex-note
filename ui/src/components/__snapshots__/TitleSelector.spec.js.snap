// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TitleSelector.vue > matches snapshot 1`] = `"<div class=\\"prime-select-stub\\" options=\\"[object Object],[object Object],[object Object]\\" optionlabel=\\"title\\" optionvalue=\\"title\\" filter=\\"\\" placeholder=\\"Select a note\\" fluid=\\"\\"></div>"`;
